shared/src/main/scala/amf/core/AMFSerializer.scala

```diff
@@ -114,31 +112,36 @@ class AMFSerializer(unit: BaseUnit,
     w.toString
   }
 
-  protected def findDomainPlugin(): Option[AMFDocumentPlugin] =
-    AMFPluginsRegistry.documentPluginForVendor(vendor).find { plugin =>
-      plugin.documentSyntaxes.contains(mediaType) && plugin.canUnparse(unit)
-    } match {
-      case Some(domainPlugin) =>
-        Some(domainPlugin)
-      case None => AMFPluginsRegistry.documentPluginForMediaType(mediaType).find(_.canUnparse(unit))
-    }
 
-  private def getDomainPlugin: AMFDocumentPlugin =
-    findDomainPlugin().getOrElse {
+  private def getRenderPlugin: AMFRenderPlugin = {
+    val renderPlugin = env.registry.plugins.renderPlugins.sorted.find(_.applies(RenderInfo(unit, vendor, mediaType)))
+    renderPlugin.getOrElse {
       throw new Exception(
         s"Cannot serialize domain model '${unit.location()}' for detected media type $mediaType and vendor $vendor")
     }
+  }
 }
 
 object AMFSerializer {
+  // this constructor will be removed as the env will be received from the client interfaces and not statically
+  def apply(unit: BaseUnit,
+            mediaType: String,
+            vendor: String,
+            options: RenderOptions,
+            shapeOptions: ShapeRenderOptions = ShapeRenderOptions()): AMFSerializer = {
+    val immutableOptions = RenderOptions.toImmutable(options)
+    val env = AMFPluginsRegistry.obtainStaticEnv().withRenderingOptions(immutableOptions)
+    new AMFSerializer(unit, mediaType, vendor, shapeOptions, env)
+  }
+
   def init()(implicit executionContext: ExecutionContext): Unit = {
```

Comment:
Can't we get rid of this `init` method?

Reply:
Would wait for when we delete RuntimeSerializer and this will become obsolete.